runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/CombineTranslation.java

```diff
@@ -181,24 +185,51 @@ public SdkFunctionSpec getCombineFn() {
         components);
   }
 
+  public static List<PCollectionView<?>> getSideInputs(AppliedPTransform<?, ?, ?> application)
+      throws IOException {
+    PTransform<?, ?> transform = application.getTransform();
+    if (transform instanceof Combine.PerKey) {
+      return ((Combine.PerKey<?, ?, ?>) transform).getSideInputs();
+    }
+
+    SdkComponents sdkComponents = SdkComponents.create();
+    RunnerApi.PTransform combineProto = PTransformTranslation.toProto(application, sdkComponents);
+    CombinePayload payload = CombinePayload.parseFrom(combineProto.getSpec().getPayload());
+
+    List<PCollectionView<?>> views = new ArrayList<>();
+    RehydratedComponents components =
+        RehydratedComponents.forComponents(sdkComponents.toComponents());
+    for (Map.Entry<String, SideInput> sideInputEntry : payload.getSideInputsMap().entrySet()) {
+      String sideInputTag = sideInputEntry.getKey();
+      RunnerApi.SideInput sideInput = sideInputEntry.getValue();
+      PCollection<?> originalPCollection =
+          checkNotNull(
+              (PCollection<?>) application.getInputs().get(new TupleTag<>(sideInputTag)),
+              "no input with tag %s",
+              sideInputTag);
+      // TODO: Should ParDoTranslation#viewFromProto live elsewhere?
```

Comment:
Yea, seems like it could like in something like `PCollectionViewTranslation`

Reply:
Moved.